File: app/src/design/java/dolphin/android/apps/CpblCalendar3/FixedRecyclerView.java
package dolphin.android.apps.CpblCalendar3;

import android.content.Context;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;
import android.util.AttributeSet;

/**
 * Created by dolphin on 2017/05/20.
 * <p>
 * http://stackoverflow.com/a/25227797
 */

public class FixedRecyclerView extends RecyclerView {
    public FixedRecyclerView(Context context) {
        super(context);
    }

    public FixedRecyclerView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public FixedRecyclerView(Context context, @Nullable AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    public boolean canScrollVertically(int direction) {
        // check if scrolling up
        if (direction < 1) {
            boolean original = super.canScrollVertically(direction);
            return !original && getChildAt(0) != null && getChildAt(0).getTop() < getPaddingTop() || original;
        }
        return super.canScrollVertically(direction);

    }
}

File: app/src/design/java/dolphin/android/apps/CpblCalendar3/FixedRecyclerView.kt
package dolphin.android.apps.CpblCalendar3

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.RecyclerView

/**
 * For RecyclerView + SwipeRefreshLayout
 *
 * See http://stackoverflow.com/a/25227797
 */

class FixedRecyclerView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null,
                                                  defStyleAttr: Int = 0)
    : RecyclerView(context, attrs, defStyleAttr) {

    override fun canScrollVertically(direction: Int): Boolean {
        // check if scrolling up
        if (direction < 1) {
            val original = super.canScrollVertically(direction)
            return !original && getChildAt(0) != null && getChildAt(0).top < paddingTop || original
        }
        return super.canScrollVertically(direction)
    }
}
